mlrun/serving/routers.py

```diff
@@ -265,39 +335,63 @@ def _resolve_route(self, body, urlpath):
             # process the url <prefix>/<model>[/versions/<ver>]/operation
             subpath = ""
             urlpath = urlpath[len(self.url_prefix) :].strip("/")
+
+            # Test if Only `self.url_prefix/` was given
             if not urlpath:
-                # Only self.url_prefix was given
                 return "", None, ""
             segments = urlpath.split("/")
+
+            # Test for router level `/operation`
             if len(segments) == 1:
+                # Path =  <prefix>/<segment>
                 # Are we looking at a router level operation?
-                if segments[0] in ["infer", "predict", "explain"]:
-                    # We are given an operation
+                try:
+                    operation = OperationTypes(segments[0])
                     self.log_router = True
-                    return self.name, None, segments[0]
-            model = segments[0]
+                    return self.name, None, operation
```

Comment:
The line you want to "catch" exceptions to is only the first one, so please change to:
 you're expecting to fail
```
                try:
                    operation = OperationTypes(segments[0])
                # Unrecognized operation was given, probably a model name
                except ValueError:
                    model = segments[0]
                else:
                    self.log_router = True
                    return self.name, None, operation